### PR TITLE
[Pods Configuration 1 of 3] Basic

### PR DESCRIPTION
**Pull request by nLight**
This PR introduces pod configuration tab view with basic sections:
Env vars, labels, version, user, secrets, volumes and scaling.

Part of the ticket: https://mesosphere.atlassian.net/browse/DCOS-9823

screen shot 2016-09-21 at 10 26 28

### CODE REVIEW

File: src/js/utils/MarathonConfigUtil.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+function buildHostName(id, port) {
+  let hostname = HostUtil.stringToHostname(id);
+  return `${hostname}${Networking.L4LB_ADDRESS}:${port}`;
+}
+
+function defaultCreateLink(contents) {
+  return contents;
+}
+
+var MarathonConfigUtil = {
```
Comment (orlandohohmeier) [inline src/js/utils/MarathonConfigUtil.js]: Let's rename this into ServiceConfigUtil as it's dealing with it's dealing with `service` structs and their properties. #nit

File: src/js/components/PodSpecView.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
+import React from 'react';
+
+import DescriptionList from './DescriptionList';
+import PodSpec from '../structs/PodSpec';
+import Icon from './Icon';
+
+const METHODS_TO_BIND = [
+];
+
+class PodSpecView extends React.Component {
```
Comment (orlandohohmeier) [inline src/js/components/PodSpecView.js]: Could you please rename this into `PodConfigurationView` or create an issue to make sure we adjust the "service" configuration view naming. Let's make sure we keep the component names in sync.

Comment (orlandohohmeier) [inline src/js/components/PodSpecView.js]: Thinking about it, I actually prefer to rename the service config view.

Comment (wavesoft) [inline src/js/components/PodSpecView.js]: Yes, I am also in favour of the latter. There is a need for some general renaming of the old *Service stuff to *Application stuff.

File: src/js/utils/MarathonConfigUtil.js (hunk 1 above)
Comment (wavesoft) [inline src/js/utils/MarathonConfigUtil.js]: Good idea, so it's aligned with the rest of the services 👍

File: src/js/utils/MarathonConfigUtil.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  getCommandString(container) {
```
Comment (orlandohohmeier) [inline src/js/utils/MarathonConfigUtil.js]: Let's please add tests to make verify that this util works as expected. Please test especially possible pod/app conflicts.

**Comment by orlandohohmeier** [comment]
Removed the "needs tests" label here as the respective functionality is introduced in #1117

File: src/js/utils/MarathonConfigUtil.js (hunk 1 above)
Comment (nLight) [inline src/js/utils/MarathonConfigUtil.js]: Implemented in #1117

File: src/js/utils/MarathonConfigUtil.js (hunk 3 above)
Comment (nLight) [inline src/js/utils/MarathonConfigUtil.js]: Implemented in #1117

File: src/js/components/PodSpecView.js (hunk 2 above)
Comment (orlandohohmeier) [inline src/js/components/PodSpecView.js]: @nLight Could you please create a Jira to track that work?

**Comment by nLight** [comment]
Rebased that one

**Comment by nLight** [comment]
@mesosphere-ci retest this please

**Comment by nLight** [comment]
@orlandohohmeier @jfurrow review this please 🙇

File: src/js/components/PodSpecView.js (hunk 2 above)
Comment (nLight) [inline src/js/components/PodSpecView.js]: @orlandohohmeier here you go https://mesosphere.atlassian.net/browse/DCOS-9919

File: src/js/components/PodDetail.js [hunk 4]
```diff
@@ -9,8 +9,9 @@ import ServiceSuspendModal from './modals/ServiceSuspendModal';
 import Pod from '../structs/Pod';
 import PodActionItem from '../constants/PodActionItem';
 import PodDebugTabView from './PodDebugTabView';
-import PodHeader from './PodHeader';
 import PodInstancesView from './PodInstancesView';
+import PodConfigurationTabView from './PodConfigurationTabView';
+import PodHeader from './PodHeader';
```
Comment (Poltergeist) [inline src/js/components/PodDetail.js]: Please sort these imports alphabetical

Comment (Poltergeist) [inline src/js/components/PodDetail.js]: Please sort these imports alphabetical.

File: src/js/components/PodSpecView.js [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+    if (Object.keys(environment).length === 0) {
+      return null;
+    }
+
+    let hash = Object.keys(environment).reduce(function (memo, key) {
+      let value = environment[key];
+
+      if (typeof value === 'object' && !Array.isArray(value) &&
+          value !== null) {
```
Comment (Poltergeist) [inline src/js/components/PodSpecView.js]: I would suggest another format here:

```JS
if (
  typeof value === 'object' && 
  !Array.isArray(value) &&
  value !== null
) {
```
This is a personal taste, feel free to ignore.

File: src/js/components/PodSpecView.js [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+          Secrets
+        </h4>
+        <DescriptionList hash={hash} />
+      </div>
+    );
+  }
+
+  getLabelsDetails() {
+    let {spec} = this.props;
+    let labels = spec.getLabels();
```
Comment (Poltergeist) [inline src/js/components/PodSpecView.js]: #nit :-)

Do not spread spec if it is used only once.

```JS
let labels = this.props.spec.getLabels();

File: src/js/components/PodSpecView.js (hunk 5 above)
Comment (nLight) [inline src/js/components/PodSpecView.js]: I like that! 
There's an idea to extract this check into a utility function like `Utils.isObject` in a separate PR of course.

**Comment by nLight** [comment]
@Poltergeist done with comments, thank you!

**Comment by nLight** [comment]
🙇

**Pull request merged**